auxx/src/Command/Tx.hs

```diff
@@ -100,62 +95,76 @@ sendToAllGenesis diffusion (SendToAllGenesisParams duration conc delay_ tpsSentF
-                          return (TxCount submitted failed (sending - 1), ())
+                      modifySharedAtomic tpsMVar $ \(TxCount submitted sending) ->
+                          return (TxCount submitted (sending - 1), ())
                 | otherwise = (atomically $ tryReadTQueue txQueue) >>= \case
-                      Just (key, txOuts) -> do
-                          utxo <- getOwnUtxoForPk $ safeToPublic (fakeSigner key)
-                          etx <- createTx mempty utxo (fakeSigner key) txOuts (toPublic key)
-                          case etx of
-                              Left err -> do
-                                  addTxFailed tpsMVar
-                                  logError (sformat ("Error: "%build%" while trying to send") err)
-                              Right (tx, _) -> do
-                                  res <- submitTxRaw diffusion tx
-                                  addTxSubmit tpsMVar
-                                  logInfo $ if res
-                                      then sformat ("Submitted transaction: "%txaF) tx
-                                      else sformat ("Applied transaction "%txaF%", however no neighbour applied it") tx
+                      Just (tx, _) -> do
+                          res <- submitTxRaw diffusion tx
+                          addTxSubmit tpsMVar
+                          logInfo $ if res
+                                    then sformat ("Submitted transaction: "%txaF) tx
+                                    else sformat ("Applied transaction "%txaF%", however no neighbour applied it") tx
                           delay $ ms delay_
                           logInfo "Continuing to send transactions."
                           sendTxs (n - 1)
-                      Nothing -> logInfo "No more transactions in the queue."
+                      Nothing -> do
+                          logInfo "No more transactions in the queue."
+                          sendTxs 0
+
             sendTxsConcurrently n = void $ forConcurrently [1..conc] (const (sendTxs n))
+        -- pre construct the first batch of transactions. Otherwise,
+        -- we'll be CPU bound and will not achieve high transaction
+        -- rates. If we pre construct all the transactions, the
+        -- startup time will be quite long.
+        forM_  firstBatch addTx
         -- Send transactions while concurrently writing the TPS numbers every
         -- slot duration. The 'writeTPS' action takes care to *always* write
         -- after every slot duration, even if it is killed, so as to
         -- guarantee that we don't miss any numbers.
-        void $ concurrently writeTPS (sendTxsConcurrently duration)
+        --
+        -- While we're sending, we're constructing the second batch of
+        -- transactions.
+        void $
+            concurrently (forM_ secondBatch addTx) $
+            concurrently writeTPS (sendTxsConcurrently txsPerThread)
```

Comment:
fwiw there's `concurrently_` that would omit the need for `void`

Reply:
`concurrently_` is not exported from Mockable module